src/Places/AbstractPlace.php

```diff
@@ -72,10 +72,40 @@ public function getThreshold()
      * @var array the failures, timeout and treshold
      *
      * @return self
+     *
+     * @throws InvalidPlaceException
      */
     public static function fromArray(array $settings)
     {
-        return new static($settings[0], $settings[1], $settings[2]);
+        static::validateSettings($settings);
```

Comment:
if you validate the settings, why do you need a ternary check after the validation?

Reply:
because I added another format of settings with associative array, the validate methods ensures one of the two allowed format is used, but then I need to know which one for instanciation
I added this new format because the initial one is not very easy to use/read, you don't know which parameter is supposed to be retry/timeout/threshold.. is it the first? the second?

Reply:
as you may bump a new major, you could keep only the second one?

Reply:
I wanted to keep the former format as well, in case it already has been integrated by another module. But I'm fine with removing the former one, I don't think many modules use this library yet.
Or maybe I could throw a deprecation warning.